tests/compile-and-dump/Singletons/StandaloneDeriving.hs

```diff
@@ -0,0 +1,32 @@
+module Singletons.StandaloneDeriving where
+
+import Data.Singletons.Prelude
+import Data.Singletons.Prelude.Show
+import Data.Singletons.TH
+
+$(singletons [d|
+
+  infixl 6 :*:
+  data T a = a :*: a
+  data S = S1 | S2 deriving Eq
```

Comment:
So `partitionDecs` would still handle standalone-deriving for non-`Eq` classes? If "yes" (which I think is reasonable) then there is no need for the full generality of standalone-deriving declarations in `PartitionedDecs`. Just add a field which is a list of names for which there is a standalone-deriving for `Eq`. Or do we need the instance contexts, too?